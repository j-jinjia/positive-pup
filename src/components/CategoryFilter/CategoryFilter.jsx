import "./CategoryFilter.scss";
import useWindowSize from "../../hooks/useWindowSize";

const CategoryFilter = ({ courseType, handleClick, filterOptions }) => {
  const unselected = "category-filter-container__filter-option";
  const selected =
    "category-filter-container__filter-option category-filter-container__filter-option--selected";

  const windowIsDesktop = useWindowSize(1024);

  const mapFiltersDesktop = filterOptions.map((val, index) => {
    return (
      <button
        key={index}
        className={courseType == val ? selected : unselected}
        onClick={handleClick}
        value={val}
      >
        {val}
      </button>
    );
  });

  const desktopFilters = (
    <div className="category-filter-container">{mapFiltersDesktop}</div>
  );

  const mapFiltersMobile = filterOptions.map((val, index) => {
    return (
      <option
        className="category-filter-dropdown-menu__option"
        key={index}
        value={val}
      >
        {val}
      </option>
    );
  });

  const mobileFilters = (
    <select
      className="category-filter-dropdown-menu"
      name="filter-options"
      onChange={handleClick}
    >
      {mapFiltersMobile}
    </select>
  );

  return <>{windowIsDesktop ? desktopFilters : mobileFilters}</>;
};

export default CategoryFilter;
