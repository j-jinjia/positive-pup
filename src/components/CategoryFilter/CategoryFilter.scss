@use "../../styles/_variables.scss" as *;

.category-filter-container {
  display: flex;
  justify-content: space-between;
  gap: 20px;

  &__filter-option {
    cursor: pointer;
    font-size: $font-size-general;
    color: $colour-quatenary;
    font-weight: $font-filter-weight;

    &--selected {
      color: $colour-font-general;
    }
  }
}

.category-filter-dropdown-menu {
  border: none;
  color: $colour-font-general;
  font-size: 1rem;
  width: max-content;
}
