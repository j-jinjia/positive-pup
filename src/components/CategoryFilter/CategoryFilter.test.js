import { customRender } from "../../utils/testUtils";
import CategoryFilter from "./CategoryFilter";

it("should match CategoryFilter component snapshot", () => {
  const { container } = customRender(
    <CategoryFilter
      courseType={"All"}
      filterOptions={[
        "All",
        "Online Courses",
        "Group Classes",
        "Bespoke Packages",
      ]}
    />
  );

  expect(container).toMatchSnapshot();
});
