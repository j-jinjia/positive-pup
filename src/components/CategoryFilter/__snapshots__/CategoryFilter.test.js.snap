// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match CategoryFilter component snapshot 1`] = `
<div>
  <div
    class="category-filter-container"
  >
    <button
      class="category-filter-container__filter-option category-filter-container__filter-option--selected"
      value="All"
    >
      All
    </button>
    <button
      class="category-filter-container__filter-option"
      value="Online Courses"
    >
      Online Courses
    </button>
    <button
      class="category-filter-container__filter-option"
      value="Group Classes"
    >
      Group Classes
    </button>
    <button
      class="category-filter-container__filter-option"
      value="Bespoke Packages"
    >
      Bespoke Packages
    </button>
  </div>
</div>
`;
